### PR TITLE
Derive multiple accounts from mnemonic

### PR DESCRIPTION
**Pull request by buberdds**
Closes https://github.com/oasisprotocol/oasis-wallet-web/issues/82

Most of this stuff was extracted or moved from "ledger". FromLedgerModal is now called MultiAccountsSelectionModal, redux part was renamed as well as it's not longer bound to ledger.

Screenshot 2022-09-06 at 14 55 46

Screenshot 2022-09-06 at 14 56 35

### CODE REVIEW

File: src/types/RootState.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export interface RootState {
   account: AccountState
   network: NetworkState
   transaction: TransactionState
-  ledger: LedgerState
+  multiAccounts: MultiAccountsState
```
Comment (lukaw3d) [inline src/types/RootState.ts:20]: I'd name it importAccounts or chooseAccountsToImport

and I'd name PR "Derive multiple accounts from mnemonic"

File: src/app/state/multiaccounts/index.ts [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
       state.accounts[index].selected = !state.accounts[index].selected
     },
-    accountsListed(state, action: PayloadAction<LedgerAccount[]>) {
+    accountsListed(state, action: PayloadAction<MultiAccountsListAccount[]>) {
       state.accounts = action.payload
     },
-    setStep(state, action: PayloadAction<LedgerStep>) {
+    setMnemonic(state, action: PayloadAction<string>) {
+      state.mnemonic = action.payload
+    },
```
Comment (lukaw3d) [inline src/app/state/multiaccounts/index.ts:35]: I don't like this. I think better solution would be for MultiAccountsSelector to take something like AddWalletPayload[], so there isn't a need to call `getAccountSigner(mnemonic, index)` twice (when enumerating and when opening) and risk a discrepancy in indexes

File: src/app/pages/OpenWalletPage/Features/MultiAccountsSelectionModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  return (
+    <Box gap="small">
+      {accounts.map((a, index) => (
+        <Account
+          index={index}
+          address={a.address}
+          balance={a.balance.available} // TODO: get total balance
+          type={WalletType.Ledger}
```
[the remark this answers is not in the document]
Comment (buberdds) [reply src/app/pages/OpenWalletPage/Features/MultiAccountsSelectionModal/index.tsx]: Mnemonic label is fixed. Account Switcher is not showing derivation path and that's how it has to work based on Tadej's feedback.

File: src/app/state/multiaccounts/index.ts (hunk 2 above)
Comment (buberdds) [reply src/app/state/multiaccounts/index.ts]: To render a selection modal we don't need public and private keys. We may not even need those keys as user can cancel an action/close modal. Question is do we want to generate private keys upfront or later when they actually needed.

File: src/app/pages/OpenWalletPage/Features/MultiAccountsSelectionModal/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/pages/OpenWalletPage/Features/MultiAccountsSelectionModal/index.tsx]: Oh, that's odd

| Importing accounts | Switching accounts |
| --- | --- |
| image image  |  image |

[the remark this answers is not in the document]
Comment (buberdds) [reply src/app/pages/OpenWalletPage/Features/MultiAccountsSelectionModal/index.tsx]: One of a next step is to show a custom account name instead.

**Review by lukaw3d — APPROVED** [review]
Great! Just split the first commit and combine some others

File: src/app/state/transaction/saga.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { call, put, race, select, take, takeEvery } from 'typed-redux-saga'
 import { ErrorPayload, ExhaustedTypeError, WalletError, WalletErrors } from 'types/errors'
 
 import { transactionActions } from '.'
-import { sign } from '../ledger/saga'
+import { sign } from '../importaccounts/saga'
```
Comment (lukaw3d) [inline src/app/state/transaction/saga.ts:10]: I didn't notice before, this ledger-specific sign function is now oddly named / oddly placed

Comment (buberdds) [reply src/app/state/transaction/saga.ts:10]: indeed, should sign and getUSBTransport generator functions be moved to `app/lib/ledger.ts` ?

Comment (lukaw3d) [reply src/app/state/transaction/saga.ts:10]: yep

maybe also have fewer ledger sign functions?
- sign from ../importaccounts/saga calls
- OasisTransaction.signUsingLedger calls
- LedgerSigner.sign calls
- OasisApp.sign

**Pull request merged**